hydromt_wflow/wflow.py

```diff
@@ -235,95 +235,245 @@ def setup_basemaps(
+        elevtn_map="wflow_dem",
+        river_routing="local-inertial",
+        land_routing="kinematic-wave",
+    ):
+        """This component adds a hydrologically conditioned elevation (hydrodem) map for
+        river and/or land local-inertial routing.
+
+        River cells are always conditioned to D8 flow directions.  Land cells are conditioned
+        to D4 flow directions if land_routing="local-inertial", else to D8.
+
+        The conditioned elevation can be based on the average cell elevation ("wflow_dem")
+        or subgrid outlet pixel elevation ("dem_subgrid"). For local-inertial river
+        routing the subgrid elevation might provide a better representation of the river
+        elevation profile, however in combination with local-inertial land routing the
+        subgrid elevation will likely overestimate the floodplain storage capacity.
+
+        Note that the same input elevation map should be used for river bankfull elevation
+        and land elevation when using local-inertial land routing.
+
+        Adds model layers:
+
+        * **hydrodem** map: hydrologically conditioned elevation [m+REF].
+
+        Parameters
+        ----------
+        elevtn_map: {"wflow_dem", "dem_subgrid"}
+            Name of staticmap to hydrologically condition, by default "wflow_dem".
+        river_routing, land_routing : {"kinematic-wave", "local-inertial"}
+            changes wflow config model.river_routing (by default "local-inertial")
+            and model.land_routing (default "kinematic-wave")
+
+        See Also
+        --------
+        hydromt.flw.dem_adjust
+        pyflwdir.FlwdirRaster.dem_adjust
+        """
+        r_list = ["kinematic-wave", "local-inertial"]
+        if not elevtn_map in self.staticmaps:
+            raise ValueError(f'"{elevtn_map}" not found in staticmaps')
+        if river_routing not in r_list:
+            raise ValueError(
+                f'river_routing="{river_routing}" unknown. Select from {r_list}.'
+            )
+        if land_routing not in r_list:
+            raise ValueError(
+                f'land_routing="{land_routing}" unknown. Select from {r_list}.'
+            )
+
+        postfix = {"wflow_dem": "_avg", "dem_subgrid": "_subgrid"}.get(elevtn_map, "")
+        name = f"hydrodem{postfix}"
+
+        self.logger.info(f"Preparing {name} map for routing.")
+        connectivity = {"local-inertial": 4, "kinematic-wave": 8}[land_routing]
+        ds_out = flw.dem_adjust(
+            da_flwdir=self.staticmaps[self._MAPS["flwdir"]],
+            da_elevtn=self.staticmaps[elevtn_map],
+            da_rivmsk=self.staticmaps[self._MAPS["rivmsk"]],
+            flwdir=self.flwdir,
+            connectivity=connectivity,
+            river_d8=True,
+            logger=self.logger,
+        ).rename(name)
+        self.set_staticmaps(ds_out)
+
+        # update toml model.river_routing
+        self.logger.debug(f'Update wflow config momdel.river_routing="{land_routing}"')
```

Comment:
@verseve Could you check if these settings L457-L475 are correct?

Reply:
Yes, this looks correct, made some small corrections (typos) in https://github.com/Deltares/hydromt_wflow/pull/63/commits/11c95e3d4c3f4d96115483150f7c43fe29032315.

Reply:
thanks!